compiler/rustc_lint/messages.ftl

```diff
@@ -604,7 +604,8 @@ lint_opaque_hidden_inferred_bound_sugg = add this bound
 lint_or_patterns_back_compat = the meaning of the `pat` fragment specifier is changing in Rust 2021, which may affect this macro
     .suggestion = use pat_param to preserve semantics
 
-lint_out_of_scope_macro_calls = cannot find macro `{$path}` in this scope
```

Comment:
I disagree with this change, "in this scope" is a meaningful and significant part of the message.
We can lookup names in many places, e.g. in a module, or in extern prelude specifically, and here were are saying where we are looking up the name in this case - in the current scope (which happens when we resolve first segments of paths).

Reply:
@petrochenkov the primary span label retains that context of the message. I feel like it is bad practice in general to have redundant information in the different parts of the error, and we already get too many complains about "too much text in the errors". This was an attempt to walk that line. In the particular case of "in this scope" you and I know what why that is meaningful, but in practice that wording ends up being more text to say something that is generally implied: if it's not on a specific other scope, it's always "this scope". Whenever we *do* have additional context from "this scope", then it does make sense to mention it ("in trait `Trait`"). If we were to actually *mention* the scope ("cannot find macro `foo` from method `bar`"), that would change my opinion on the use of this (because then the message can be read in isolation without having to read the code to understand the context).

Reply:
I agree that "in this scope" is pretty general, and replacing it with "from this specific place called `foo`" when possible would be better, but it's definitely better than nothing because it still says which resolution mode is used.
If it was entirely useless, then it could be dropped from the label messages as well.

(As for the redundancy, I've answered below.)